python/lsst/ci/prepare.py

```diff
@@ -384,6 +384,7 @@ def _fetch(self, product):
             #     git.fetch("origin", "--force", "--tags")
             # but avoids the overhead of two (possibly remote) git calls.
             git.fetch("-fup", "origin", "+refs/heads/*:refs/heads/*", "refs/tags/*:refs/tags/*")
```

Comment:
I see that the `-u` option is documented as "don't use this".

Reply:
Nice catch on the manpage.  `git fetch -p && git fetch -p` doesn't result in the local ref being removed / broken `HEAD` -- the `-u` seems to be required on the first call to result in this behavior.